packages/react-components/react-charts-preview/src/components/LineChart/LineChart.tsx

```diff
@@ -1,16 +1,1312 @@
+    let _yAxisScale: any = '';
+    let _circleId: string = getId('circle');
+    let _lineId: string = getId('lineID');
+    let _borderId: string = getId('borderID');
+    let _verticalLine: string = getId('verticalLine');
+    let _colorFillBarPatternId: string = getId('colorFillBarPattern');
+    let _uniqueCallOutID: string | null = '';
+    let _refArray: IRefArrayData[] = [];
+    let margins: IMargins;
+    let eventLabelHeight: number = 36;
+    let lines: JSX.Element[];
+    let _renderedColorFillBars: JSX.Element[];
+    let _colorFillBars: IColorFillBarsProps[] = [];
+    let _tooltipId: string = getId('LineChartTooltipId_');
+    let _rectId: string = getId('containerRectLD');
+    let _staticHighlightCircle: string = getId('staticHighlightCircle');
+    let _createLegendsMemoized: (data: LineChartDataWithIndex[]) => JSX.Element = memoizeFunction(
+      (data: LineChartDataWithIndex[]) => _createLegends(data),
+    );
+    let _firstRenderOptimization: boolean = true;
+    let _emptyChartId: string = getId('_LineChart_empty');
+    const _isRTL: boolean = isRtl();
+    let xAxisCalloutAccessibilityData: IAccessibilityProps;
+
+    const [hoverXValue, setHoverXValue] = React.useState<string | number>('');
+    const [activeLegend, setActiveLegend] = React.useState<string>('');
+    const [YValueHover, setYValueHover] = React.useState<[]>([]);
+    const [refSelected, setRefSelected] = React.useState<SVGGElement | null | string | undefined>(null);
+    const [selectedLegend, setSelectedLegend] = React.useState<string>('');
+    const [isCalloutVisible, setIsCalloutVisible] = React.useState<boolean>(false);
+    const [selectedLegendPoints, setSelectedLegendPoints] = React.useState<any[]>([]);
+    const [selectedColorBarLegend, setSelectedColorBarLegend] = React.useState<any[]>([]);
+    const [isSelectedLegend, setIsSelectedLegend] = React.useState<boolean>(false);
+    const [activePoint, setActivePoint] = React.useState<string>('');
+    const [nearestCircleToHighlight, setNearestCircleToHighlight] = React.useState<ILineChartDataPoint | null>(null);
+    const [activeLine, setActiveLine] = React.useState<number>(0);
+    const [dataPointCalloutProps, setDataPointCalloutProps] = React.useState<ICustomizedCalloutData>();
+    const [stackCalloutProps, setStackCalloutProps] = React.useState<ICustomizedCalloutData>();
+
+    const pointsRef = React.useRef([]);
+    const calloutPointsRef = React.useRef([]);
+    React.useEffect(() => {
+      /** note that height and width are not used to resize or set as dimesions of the chart,
+       * fitParentContainer is responisble for setting the height and width or resizing of the svg/chart
+       */
+      if (_points !== _injectIndexPropertyInLineChartData(props.data.lineChartData) || props.data !== _points) {
+        pointsRef.current = _injectIndexPropertyInLineChartData(props.data.lineChartData);
+        calloutPointsRef.current = calloutData(pointsRef.current);
+      }
+    });
```

Comment:
add a effect trigger condition of change in height, weight and data

Reply:
Okay